src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -259,6 +259,27 @@ private ConstantValue DefaultSyntaxValue
                         if (!_lazyDefaultSyntaxValue.IsBad)
                         {
                             VerifyParamDefaultValueMatchesAttributeIfAny(_lazyDefaultSyntaxValue, parameterEqualsValue.Value.Syntax, diagnostics);
+
+                            // Ensure availability of `*ConstantAttribute`.
+                            // Ignoring lambdas to avoid duplicate errors with `Binder.BindToInferredDelegateType`.
+                            if (DefaultValueFromAttributes == ConstantValue.NotAvailable &&
+                                this is not LambdaParameterSymbol)
```

Comment:
>this is not LambdaParameterSymbol [](http://example.com/codeflow?start=32&length=33)

Why do we have this condition? What if this lambda has nothing to do with synthesizing a delegate? Are we going to completely skip the check?